tests/Aql/ExecutionBlockImplTest.cpp

```diff
@@ -391,6 +401,1167 @@ TEST_F(ExecutionBlockImplTest,
+   */
+  ProduceCall generateNeverProduceCall() {
+    return [](AqlItemBlockInputRange& input,
+              OutputAqlItemRow& output) -> std::tuple<ExecutorState, LambdaExe::Stats, AqlCall> {
+      // Should not be called here. No limit, only skip!
+      EXPECT_TRUE(false);
+      THROW_ARANGO_EXCEPTION(TRI_ERROR_DEBUG);
+    };
+  }
+};
+
+/**
+ * @brief Test the internal statemachine of the ExecutionBlockImpl.
+ *        These test-cases focus on a single executor and assert that this Executor is called
+ *        correctly given an input.
+ *        This is a parameterized test and tests passthrough (true) and non-passthrough variants (false)
+ */
+class ExecutionBlockImplExecuteSpecificTest : public SharedExecutionBlockImplTest,
+                                              public testing::TestWithParam<bool> {
+ protected:
+  /**
+   * @brief Generic test runner. Creates Lambda Executors, and returns ExecutionBlockImpl.execute(call),
+   *
+   * @param prod The Produce call that should be used within the Lambda Executor
+   * @param skip The Skip call that should be used wiithin the Lambda Executor (only used for non-passthrough)
+   * @param call The AqlCall that should be applied on the Executors.
+   * @return std::tuple<ExecutionState, size_t, SharedAqlItemBlockPtr>  Response of execute(call);
+   */
+  auto runTest(ProduceCall& prod, SkipCall& skip, AqlCall call)
+      -> std::tuple<ExecutionState, size_t, SharedAqlItemBlockPtr> {
+    AqlCallStack stack{std::move(call)};
+    auto singleton = createSingleton();
+    if (GetParam()) {
+      ExecutionBlockImpl<LambdaExePassThrough> testee{fakedQuery->engine(),
+                                                      generateNodeDummy(),
+                                                      makeInfos(prod)};
+      testee.addDependency(singleton.get());
+      return testee.execute(stack);
+    } else {
+      ExecutionBlockImpl<LambdaExe> testee{fakedQuery->engine(), generateNodeDummy(),
+                                           makeSkipInfos(prod, skip)};
+      testee.addDependency(singleton.get());
+      return testee.execute(stack);
+    }
+  }
+};
+
+// Test a default call: no skip, no limits.
+TEST_P(ExecutionBlockImplExecuteSpecificTest, test_toplevel_unlimited_call) {
+  AqlCall fullCall{};
+  size_t nrCalls = 0;
+
+  // Note here: passthrough only reserves the correct amount of rows.
+  // As we fetch from a singleton (1 row) we will have 0 rows (cold-start) and then exactly 1 row
+  // in the executor.
+  // Non passthrough does not make an estimate for this, so Batchsize is used.
+  ProduceCall execImpl = GetParam() ? generateProduceCall(nrCalls, fullCall, 0, 1)
+                                    : generateProduceCall(nrCalls, fullCall);
+  SkipCall skipCall = generateNeverSkipCall();
+  auto [state, skipped, block] = runTest(execImpl, skipCall, fullCall);
+
+  EXPECT_EQ(state, ExecutionState::DONE);
+  EXPECT_EQ(skipped, 0);
+  EXPECT_EQ(block, nullptr);
```

Comment:
Shouldn't this be a block with a single, empty line? I have the same question for the test cases below.

Reply:
Sorry that was misleading, we do not produce rows by the above block
So we assert that we will get a nullptr back.
We assert on the Calls here, data forwarding is in integration test suite.